### DISCUSSION TITLE
Hardware Timer Example

### DISCUSSION THREAD

**Discussion by DinkarY** [zephyrproject-rtos/zephyr#70151] [General]
Hello,

I am using "esp32_devkitm".
I want a timer interrupt for 40 microseconds for my application. For that, I want to use a Hardware Timer. However, I found no example or reference document that helped me to create a hardware timer code.

can anyone tell me how to create a Hardware timer code?

Thanks,
Dinkar

**Comment by RichardSWheatley** [comment]
ESP has documentation on their website.
https://docs.espressif.com/projects/esp-idf/en/v4.3/esp32/api-reference/peripherals/timer.html
https://github.com/espressif/esp-idf/tree/master/examples/peripherals/timer_group/gptimer

If you wanted to, you could create a Zephyr example from this information.

**Comment by DinkarY** [comment]
Hello, 
Thanks for the suggestion.
Can I use the same timer-related API of ESP-IDF to create the Zephyr project?
Is there any specific setting I have to add in the Zephyr project to support the hardware timer of esp32?
is there any guide to port code from ESP-IDF to zephyr?

Thanks,
Dinkar

**Reply by RichardSWheatley** [reply]
I think you would need to check and see if HW Timers are setup in Zephyr for ESP.

If not, you could add it or you could contact the company that makes that chipset. Zephyr is a open-source community so adding code is up to the person who needs it or the company that supports it.

It's probably not going to be a 1-1 correlation and I have not seen any porting guide.

**Reply by rftafas** [reply]
Timers are already supported. The support for S3, for example, was made [here](https://github.com/zephyrproject-rtos/zephyr/pull/55732/files).

Looking into this PR, you can take an idea of some overlays to use it.

**Reply by aslansq** [reply]
![Screenshot from 2025-01-21 19-00-28](https://github.com/user-attachments/assets/62fd4b16-fd69-4bb7-943c-7b340da1368f)
I went to rabbit hole, wrote to register interrupt does not trigger for some reason
timer counts I could see that
I dont have any clue

**Reply by aslansq** [reply]
https://github.com/aslansq/zephyr/pull/1/files

I have shared what I have tried if I anyone knows my mistake
